src/DynamoCoreWpf/Views/HomePage/HomePage.xaml.cs

```diff
@@ -0,0 +1,559 @@
+            }
+
+            if (this.startPage.DynamoViewModel.OpenCommand.CanExecute(path))
+                this.startPage.DynamoViewModel.OpenCommand.Execute(path);
+        }
+
+        internal void StartGuidedTour(string path)
+        {
+            if (String.IsNullOrEmpty(path)) return;
+            if (DynamoModel.IsTestMode)
+            {
+                TestHook?.Invoke(path);
+                return;
+            }
+
+            ShowGuidedTour(path);
+        }
+
+        internal void NewWorkspace()
+        {
+            this.startPage?.DynamoViewModel?.NewHomeWorkspaceCommand.Execute(null);
+        }
+
+        internal void OpenWorkspace()
+        {
+            if (DynamoModel.IsTestMode)
+            {
+                TestHook?.Invoke(string.Empty);
+                return;
+            }
+
+            this.startPage?.DynamoViewModel?.ShowOpenDialogAndOpenResultCommand.Execute(null);
+        }
+
+        internal void NewCustomNodeWorkspace()
+        {
+            if (DynamoModel.IsTestMode)
+            {
+                TestHook?.Invoke(string.Empty);
+                return;
+            }
+
+            this.startPage?.DynamoViewModel?.ShowNewFunctionDialogCommand.Execute(null);
+        }
+
+        internal void ShowSampleFilesInFolder()
+        {
+            if (this.startPage == null) return;
+            if (DynamoModel.IsTestMode)
+            {
+                TestHook?.Invoke(string.Empty);
+                return;
+            }
+
+            Process.Start(new ProcessStartInfo("explorer.exe", "/select,"
+                + this.startPage.SampleFolderPath)
+            { UseShellExecute = true });
+        }
+
+        internal void ShowBackupFilesInFolder()
+        {
+            if (this.startPage == null) return;
+            if (DynamoModel.IsTestMode)
+            {
+                TestHook?.Invoke(string.Empty);
+                return;
+            }
+
+            Process.Start(new ProcessStartInfo("explorer.exe", this.startPage.DynamoViewModel.Model.PathManager.BackupDirectory)
+            { UseShellExecute = true });
+        }
+
+        internal void ShowTemplate()
+        {
+            if (DynamoModel.IsTestMode)
+            {
+                TestHook?.Invoke(string.Empty);
+                return;
+            }
+
+            // Equivalent to CommandParameter="Template"
+            this.startPage?.DynamoViewModel?.ShowOpenTemplateDialogCommand.Execute("Template"); 
+        }
+
+        internal void ApplicationLoaded()
+        {
+            LoadingDone();  
+        }
+
+        #endregion
+
+        #region Dispose
+        public void Dispose()
+        {
+            DataContextChanged -= OnDataContextChanged;
+            if(startPage != null) startPage.DynamoViewModel.PropertyChanged -= DynamoViewModel_PropertyChanged;
+
+
+            if (File.Exists(fontFilePath))
+            {
+                File.Delete(fontFilePath);
+            }
+        }
```

Comment:
this does not look like a full implementation of the Dispose pattern. We're not suppressing the finalizer and we're not cleaning up the webview2 control.
Any reason for it ?

Reply:
@dnenov ?